### DISCUSSION TITLE
I am not sure its bug or wrong coding AlertDoes not popUp

### DISCUSSION THREAD

**Discussion by saroar** [pointfreeco/swift-composable-architecture#702] [Q&A]
I am just sharing with little code I hope it is all clear if not I will create a new app 
1st problem alert does not pop
 
```
public let eventFormReducer = Reducer<
  EventFormState, EventFormAction, EventFormEnvironment
> { state, action, environment in
 case . alertDismissed:
  // wanted to go my parent view
 // so i think i can make child state from my parent it will back to parentView
  return .none
  case .successFullMsg: // 
    state.alert = .init(
      title: .init("Event: \(state.title)"),
      message: .init("Your event \(state.title) successfully created!"),
      primaryButton: .cancel(send: .alertDismissed),
      secondaryButton: .default(.init("Dismiss"), send: .alertDismissed)
    )

    state.isPostRequestOnFly = false
    return .none

  case let .eventsResponse(.success(event)):

    return Effect(value: EventFormAction.successFullMsg)
      .delay(for: 1, scheduler: environment.mainQueue)
      .eraseToEffect()

  case let .eventsResponse(.failure(error)):
    state.isPostRequestOnFly = false
    return .none

  case .submitButtonTapped:
       return .envirionment.....with data 
}
<img width="1237" alt="Screenshot 2021-08-07 at 13 59 07" src="https://user-images.githubusercontent.com/8770772/128598053-032ca368-9ec3-4d56-852c-183de3d00a6d.png">

Environment

Xcode 13 beta 3
Swift 5.5
OS (if applicable): iOS 15

**Comment by stephencelis** [comment]
@saroar Unfortunately without more code (ideally a working sample that reproduces the issue) it'll be tough for anyone to troubleshoot. A few questions along the way, though:

- Have you set breakpoints in the reducer or used one of our debug helpers to ensure that alert state is getting set?
- What does the view code look like that attempts to display the alert?

This doesn't sound like a bug in the library, especially since we have working case studies showing how alerts can be displayed in TCA, so I'm going to convert this issue to a discussion. If it _does_ turn out to be a bug in the library, we can open up another one here with more information.

**Reply by saroar** [reply]
I think you answer me already asking this quis `What does the view code look like that attempts to display the alert?` thanks 
I was missing an alert in my view
